python/tank/authentication/defaults_manager.py

```diff
@@ -87,13 +90,24 @@ def get_http_proxy(self):
         Called by the authentication system when it needs to retrieve the
         proxy settings to use for a Shotgun connection.
 
-        The format should be the same as is being used in the Shotgun API.
-        For more information, see the Shotgun API documentation:
-        https://github.com/shotgunsoftware/python-api/wiki/Reference%3A-Methods#shotgun
+        .. note:: If the centralized settings do not specify an HTTP proxy, Toolkit will rely on
+            Python's `urllib.getproxies <https://docs.python.org/2/library/urllib.html#urllib.getproxies>`_
+            to find an HTTP proxy.
+
+            There is a restriction when looking for proxy information from Mac OS X System Configuration or
+            Windows Systems Registry: in these cases, Toolkit does not support the use of proxies
+            which require authentication (username and password).
+
+        The returned format will be the same as is being used in the Shotgun API.
+        For more information, see the `Shotgun API documentation
+        <http://developer.shotgunsoftware.com/python-api/reference.html#shotgun>`_.
 
         :returns: String containing the default http proxy, None by default.
         """
-        return self._user_settings.shotgun_proxy
+        if self._user_settings.shotgun_proxy is None:
```

Comment:
This is the new magic. If the value is not set, default to the system settings. Otherwise use that value.